go/booking_test.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/mwinteringham/api-framework/go/api"
+	"github.com/mwinteringham/api-framework/go/payloads"
+	"github.com/stretchr/testify/assert"
+)
+
+type createReqFunc func(payloads.Booking) (*http.Request, error)
+type updateReqFunc func(payloads.Booking, string) (*http.Request, error)
+
+func TestAllBookings(t *testing.T) {
+	req, err := api.RequestAllBookings()
+	if err != nil {
+		fmt.Println(err)
+	}
+	statusCode, _ := DoRequest(req)
+	assert.Equal(t, 200, statusCode)
+}
+
+func TestSpecificBooking(t *testing.T) {
+	req, err := api.RequestSpecificBooking("1")
+	if err != nil {
+		fmt.Println(err)
+	}
+	statusCode, _ := DoRequest(req)
+	assert.Equal(t, 200, statusCode)
+}
+
+func TestCreateBookings(t *testing.T) {
+	validBooking := payloads.Booking{
+		Firstname:   "Jim",
+		Lastname:    "Brown",
+		TotalPrice:  111,
+		DepositPaid: true,
+		BookingDates: payloads.BookingDates{
+			CheckIn:  "2018-01-01",
+			CheckOut: "2019-01-01",
+		},
+	}
+
+	validBookingWithBreakfast := payloads.Booking{
+		Firstname:   "Jim",
+		Lastname:    "Brown",
+		TotalPrice:  111,
+		DepositPaid: true,
+		BookingDates: payloads.BookingDates{
+			CheckIn:  "2018-01-01",
+			CheckOut: "2019-01-01",
+		},
+		AdditionalNeeds: "Breakfast",
+	}
+
+	invalidBooking := payloads.Booking{
+		TotalPrice: 111,
+	}
+
+	tests := []struct {
+		reqFunc            createReqFunc
+		payload            payloads.Booking
+		ExpectedStatusCode int
+	}{
+		{api.RequestCreateBooking, validBooking, 200},
+		{api.RequestCreateBooking, validBookingWithBreakfast, 200},
+		{api.RequestCreateBooking, invalidBooking, 500},
+	}
+
+	for _, test := range tests {
+		req, err := test.reqFunc(test.payload)
+
+		if err != nil {
+			fmt.Println(err)
+		}
+		statusCode, _ := DoRequest(req)
+		assert.Equal(t, test.ExpectedStatusCode, statusCode)
+	}
+}
+
+func TestUpdateBooking(t *testing.T) {
```

Comment:
To keep the example checks in line with the other frameworks and help readers compare. Could this check focus on 'Deleting a booking returns 201'

Reply:
Delete should be 204 though, right?

Reply:
Yeah but restful-booker is a buggy piece of cr*p. I don't know who built it!